### PR TITLE
[app_dart] Track GitHub API quota usage

### PR DESCRIPTION
**Pull request by CaseyHillers**
* Add `/api/public/github-rate-limit-status` to track all supported Flutter repo API usage
  * Add 1 minute cron job to push to BigQuery (metric collection)
* Clean up of imports/exports in `bin/server.dart`

### CODE REVIEW

File: app_dart/lib/src/request_handlers/github_rate_limit_status.dart [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../foundation/utils.dart';
+import '../request_handling/body.dart';
+import '../request_handling/request_handler.dart';
+import '../service/github_service.dart';
+
+/// Endpoint to collect the current GitHub API quota usage of Cocoon on all supported Flutter repos.
```
Comment (godofredoc) [inline app_dart/lib/src/request_handlers/github_rate_limit_status.dart:15]: Can we add documentation on the format of the data being collected? I'm particularly interested on how are we managing overlapping windows.

Comment (CaseyHillers) [reply app_dart/lib/src/request_handlers/github_rate_limit_status.dart:15]: Added documentation on the BQ entry format. Overlapping windows is managed by locking this behind a CacheRequestHandler to ensure its only run once a minute.

In general, BQ documentation in Cocoon is lacking. I created https://github.com/flutter/flutter/issues/75959 to track.

Comment (godofredoc) [reply app_dart/lib/src/request_handlers/github_rate_limit_status.dart:15]: Do we still need to align the windows with the server? usually the APIs provide the necessary information in the form of headers.

**Review by godofredoc — APPROVED** [review]

File: app_dart/lib/src/request_handlers/github_rate_limit_status.dart (hunk 1 above)
Comment (CaseyHillers) [reply app_dart/lib/src/request_handlers/github_rate_limit_status.dart:15]: There's an explicit rate limit api call that's being used here. I opted for this as it was simpler and didn't see a need for monitoring more fine grained than a minute.

**Pull request merged**